src/EStimLibrary/Core/ResourceManager.cs

```diff
@@ -8,13 +8,15 @@ public class ResourceManager<ResourceType>
 
     public readonly int MaxNumResources;
 
-    public int NumTotalResources => this.IdPool.NumUsedIds;
+    public int NumTotalResources => this.Resources.Count;
 
+    // Constructor
     public ResourceManager(int baseId = 0, int initialNumResourceIds = 0,
         int maxNumResources = Constants.POS_INFINITY)
     {
         this.IdPool = new(baseId, initialNumResourceIds);
         this.Resources = new();
+        if (maxNumResources < 0) maxNumResources = Constants.POS_INFINITY;
```

Comment:
Ah I forgot I made a lib const for max int value later. Use Constants.POS_INFINITY in ReusableIdPool to wherever int.MaxValue is used right not